packages/rainbowkit/src/transactions/TransactionStoreContext.tsx

```diff
@@ -39,6 +56,12 @@ export function TransactionStoreProvider({
     }
   }, [store, address, chainId]);
 
+  useEffect(() => {
+    if (store && address && chainId) {
+      return store.onTransactionStatus(onTransactionStatus);
+    }
+  }, [store, address, chainId, onTransactionStatus]);
```

Comment:
We could pass this logic in [here](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/transactions/useRecentTransactions.ts#L20-L22) which would be less work, but there are some issues about doing that.

-Since `useBalance` from wagmi doesn't cause a automatic re-render when a transaction is confirmed. This means we'll have to do something on our end to cause a re-render and if we do it with [notifyListeners](https://github.com/rainbow-me/rainbowkit/blob/d5c3f2415b1e98b83b0f533e64e235ed6d980074/packages/rainbowkit/src/transactions/transactionStore.ts#L196) it would cause a lot of re-render to our `useRealtimeBalance` hook.
- We'll separate this logic to use our on `onTransactionStatus` event which will only pass us a status for a transaction when it's needed and when it's neccesarry. This way we'll cause a re-render to `useRealtimeBalance` and it won't waste too much RPC calls either.